paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsItem.kt

```diff
@@ -35,8 +31,6 @@ internal sealed class PaymentOptionsItem {
         val paymentMethod = displayableSavedPaymentMethod.paymentMethod
         val isModifiable: Boolean by lazy { displayableSavedPaymentMethod.isModifiable() }
 
-        override val isEnabledDuringEditing: Boolean = true
```

Comment:
I'm removing these for now, but it's clear there needs to be some `Interactor` that converts these to tabs to their expected states rather than manage the enabled and edit logic inside of UI.

Reply:
What do we lose by removing this? Is it risky to merge without your follow up?

Reply:
We were mainly using this to decide whether items should be enabled when editing which was simple before since payment methods were always removable. This allowed us to have one line in UI for deciding if something should be enabled.
```
val isEnabled = !isProcessing && (!isEditing || item.isEnabledDuringEditing)
```

With `CustomerSession`, we can now disable removal but still allow modification which means saved payment methods can now be disabled in edit mode in certain states.

There is a path for keeping this check but it forces every `PaymentOptionsItem.SavedPaymentMethod` type to b e aware of whether payment methods can be removed which I've had pushback against doing.

```
sealed interface PaymentOptionsItem {
    data class SavedPaymentMethod(
        val method: DisplayableSavedPaymentMethod,
        private val canRemovePaymentMethods: Boolean,
    ) : PaymentOptionsItem {
        override val isEnabledDuringEditing: Boolean = method.isModifiable || canRemovePaymentMethods
    }
}
```